packages/devtools_app/lib/src/version.dart

```diff
@@ -15,7 +16,21 @@ class FlutterVersion {
     @required this.frameworkCommitDate,
     @required this.engineRevision,
     @required this.dartSdkVersion,
-  });
+  }) {
+    // Flutter versions can come in as '1.10.7-pre.42', so we strip out any
+    // characters that are not digits. We do not currently have a need to know
+    // more version parts than major, minor, and patch. If this changes, we can
+    // add support for the extra values.
+    final _versionParts = version
+        .split('.')
+        .map((part) => String.fromCharCodes(
+            part.codeUnits.where((cu) => isDigit(cu)).toList()))
+        .toList();
+    major =
```

Comment:
nit: it is probably cleaner to use a regex here instead.
e.g. 
RegExp('[0-9]+') is probably what you are looking for.
That would also treat
7-pre2 as
7
instead of
72 which is probably what makes more sense.